### PR TITLE
Bug fix/internal issue #605

### PR DESCRIPTION
**Pull request by gnusi**
- release tick before committing changes to iresearch

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5114/

File: arangod/IResearch/IResearchLink.cpp [hunk 1]
```diff
@@ -1364,17 +1381,17 @@ arangodb::Result IResearchLink::initDataStore(InitCallback const& initCallback,
...
 
       // before commit ensure that the WAL 'Flush' marker for the opened writer
       // was seen, otherwise this indicates a lost WAL tail during recovery
       // i.e. dataStore is ahead of the WAL
       if (RecoveryState::AFTER_CHECKPOINT != link->_dataStore._recovery) {
-        return arangodb::Result( // result
-          TRI_ERROR_INTERNAL, // code
-          std::string("failed to find checkpoint after finishing recovery of arangosearch link '") + std::to_string(link->id()) + "'"
-        );
+        LOG_TOPIC("31zv1", ERR, arangodb::iresearch::TOPIC)
```
Comment (jsteemann) [inline arangod/IResearch/IResearchLink.cpp]: `31zv1` is an invalid log id. The rules are:
* 5 digits
* from the range `[0-9a-f]`
* globally unique

File: arangod/IResearch/IResearchLink.cpp [hunk 2]
```diff
@@ -1401,7 +1418,7 @@ arangodb::Result IResearchLink::insert( // insert document
   if (!trx.state()) {
     return arangodb::Result(
       TRI_ERROR_BAD_PARAMETER,
```
Comment (jsteemann) [inline arangod/IResearch/IResearchLink.cpp]: I think this is not a `TRI_ERROR_BAD_PARAMETER` (as this is normally used for bad parameters submitted by end users) but this justifies a `TRI_ERROR_INTERNAL` (an error that is mostly likely due to a programming error that the end user can do nothing about).

File: arangod/Utils/FlushThread.cpp [hunk 3]
```diff
@@ -67,10 +68,13 @@ void FlushThread::run() {
...
       }
 
-      flushFeature->releaseUnusedTicks(count);
+      flushFeature->releaseUnusedTicks(count, tick);
 
       LOG_TOPIC_IF("2b2h1", DEBUG, arangodb::Logger::FLUSH, count)
-          << count << " flush subscription(s) released";
+          << "Flush subscription(s) released: '" << count;
+
+      LOG_TOPIC("2b2h2", DEBUG, arangodb::Logger::FLUSH)
```
Comment (jsteemann) [inline arangod/Utils/FlushThread.cpp]: this is also going to be an invalid log topic id.

File: arangod/RestServer/FlushFeature.cpp [hunk 4]
```diff
@@ -57,42 +57,39 @@ namespace arangodb {
...
-     _engine(engine),
      _tickCurrent(0), // default (smallest) tick for StorageEngine
      _tickPrevious(0), // default (smallest) tick for StorageEngine
      _type(type) {
-    // it's too early to use _engine.currentTick() here,
-    // storage engine may not be initialized yet
   }
 
   void resetCurrentTick(TRI_voc_tick_t tick) noexcept {
     _tickPrevious = _tickCurrent;
```
Comment (jsteemann) [inline arangod/RestServer/FlushFeature.cpp]: Just wondering if this needs to be an atomic operation, or at least somehow protected against concurrent reads and writes of `_tickPrevious` and `_tickCurrent`. At least the `tick()` method is public and returns `_tickPrevious`. Is it possible that one thread is in `resetCurrentTick?` while another thread is in `tick()`?

Comment (gnusi) [reply arangod/RestServer/FlushFeature.cpp]: good point

**Pull request merged**